docs/developer-guide/etendo-classic/concepts/data-model.md

```diff
@@ -0,0 +1,513 @@
+
+| Search name           | Referred table     |
+| --------------------- | ------------------ |
+| Account               | C_ValidCombination |
+| Business Partner      | C_BPartner         |
+| Debt/Payment          | C_Debt_Payment     |
+| Invoice               | C_Invoice          |
+| Invoice Line          | C_InvoiceLine      |
+| Location              | C_Location         |
+| Locator               | M_Locator          |
+| Order                 | C_Order            |
+| Order Line            | C_OrderLine        |
+| Product               | M_Product          |
+| Product Complete      | M_Product          |
+| Project               | C_Project          |
+| Shipment/Receipt      | M_InOut            |
+| Shipment/Receipt Line | M_InOutLine        |
+
+##### Password (not decryptable)
+
+Columns with this reference are stored in database as a hash of the value the user entered; although it is not possible to recover the original plain text value, it can be checked whether a plain text corresponds to a hashed value.
+
+SHA-512 with a random salt is used to hash passwords. Previously, SHA-1 was used.
+
+To verify a plain text matches a hash (this is backwards compatible and checks both algorithms), use:  
+boolean matches = PasswordHash.matches(plainTextPassword, hashedPassword);
+
+To generate a new hash use:
+String newPassword = PasswordHash.generateHash(plainText);
+
+##### Password (decryptable)
+
+The value of the password is encrypted when being saved, however in a way that the clear text password (as entered) can be recovered. Example use-case is to store a password which needs to be passed later as clear text to some external service (i.e. a email server password).
+
+#### References in modules
+
+It is allowed for modules to define base references.
+
+Prior to this, release modules only could define subreferences for the base ones provided in core, this means that new **List**, **Table** and **Selector** references could be added to modules.
+
+### Validations
+
+Validations allow to filter the records that appear in generated drop down lists. Validations are applicable to the following data references **List**, **Table**, **TableDir** and **Button**.
+
+When a column has a validation, WAD generates a [callout](#callout) for it (**ComboReload**) to implement the validation.
+
+#### Defining a validation
+
+Validations are defined in `Application Dictionary` > `Setup` > `Validation Setup` > `Validation` , they consist in a **where** sql clause that will be applied to the query that generates and retrieves the data for the drop down list.
+
+It is possible to do reference to columns within the foreign table in the code of the validation, to do it include the full table name. In case the reference is **List** type the table is **AD_Ref_List** and the column where values are stored in is **Value**.
```

Comment:
```suggestion
It is possible to do reference to columns within the foreign table in the code of the validation, to do it include the full table name. In case the reference is **List** type the table is `AD_Ref_List` and the column where values are stored in is **Value**.
```